orttraining/orttraining/training_api/onnxruntime_training_c_api.cc

```diff
@@ -210,35 +249,21 @@ ORT_API_STATUS_IMPL(OrtTrainingApis::OptimizerStep, _Inout_ OrtTrainingSession*
   API_IMPL_END
 }
 
-ORT_API_STATUS_IMPL(OrtTrainingApis::RegisterLRScheduler, _Inout_ OrtTrainingSession* sess,
-                    _In_ void* lr_scheduler_parameters, _In_ enum OrtLRSchedulerType lr_scheduler_type,
-                    _In_opt_ const float* initial_lr) {
+ORT_API_STATUS_IMPL(OrtTrainingApis::RegisterLinearLRScheduler, _Inout_ OrtTrainingSession* sess,
+                    _In_ const int64_t warmup_step_count,
+                    _In_ const int64_t total_step_count,
+                    _In_ const float initial_lr) {
   API_IMPL_BEGIN
 
   OrtStatus* status = nullptr;
 
   auto session = reinterpret_cast<onnxruntime::training::api::TrainingSession*>(sess);
-
-  if (!lr_scheduler_parameters) {
-    return OrtApis::CreateStatus(ORT_FAIL, "The provided learning rate scheduler parameters is a nullptr.");
-  }
-
-  switch (lr_scheduler_type) {
-    case OrtLRSchedulerType::LinearLRScheduler: {
-      auto parameters = reinterpret_cast<OrtLinearLRSchedulerParameters*>(lr_scheduler_parameters);
-      ORT_API_RETURN_IF_STATUS_NOT_OK(
-          session->RegisterScheduler([&parameters](auto optimizer) {
-            return std::make_unique<onnxruntime::training::api::LinearLRScheduler>(
-                optimizer, parameters->warmup_step_count, parameters->total_step_count);
-          },
-                                     initial_lr ? std::optional<float>(*initial_lr) : std::nullopt));
-      break;
-    }
-    default: {
-      status = OrtApis::CreateStatus(ORT_FAIL, "Could not decipher the OrtLRSchedulerType from the given argument.");
-      break;
-    }
-  }
+  ORT_API_RETURN_IF_STATUS_NOT_OK(
+      session->RegisterScheduler([&](auto optimizer) {
```

Comment:
we should probably capture the primitives and optional as values and not as refs.